landice/output_processing_li/ismip6_postprocessing/process_flux_variables.py

```diff
@@ -0,0 +1,446 @@
+        if len(index_cf>0):
+            thresholdBoundary = np.zeros((nCells,), 'i')
+            thresholdBoundaryAssignedVolume = np.zeros((nCells,))
+            thresholdBoundarySummedThickness = np.zeros((nCells,))
+            thresholdBoundaryContributors = np.zeros((nCells,))
+            thresholdBoundaryLength = np.zeros((nCells,))
+            # First make list of boundary cells calved
+            for i in index_cf:
+                ne = nEdgesOnCell[i]
+                for j in range(ne):
+                    neighborCellId = cellsOnCell[i, j] - 1
+                    if thickness[t,neighborCellId] > 0.0 and bed[neighborCellId] < 0.0 and calvingThicknessFromThreshold[t,neighborCellId] == 0.0:
+                        thresholdBoundary[i] = 1
+                        thresholdBoundaryLength[i] += dvEdge[edgesOnCell[i,j]-1]
+            bdyIndices = np.where(thresholdBoundary == 1)[0]
+            print(f"Found {len(index_cf)} cells with threshold calving at time {t}; {len(bdyIndices)} are boundary cells.")
+            # Now loop over all threshold cells and assign their volume to the nearest boundary cell
+            for i in index_cf:
+                if thresholdBoundary[i] == 1:
+                    ownerIdx = i # often the cell is its own owner, so check before doing the more expensive search
+                else:
+                    ownerIdx = bdyIndices[np.argmin((xCell[i]-xCell[bdyIndices])**2 + (yCell[i]-yCell[bdyIndices])**2)]
+                thresholdBoundaryAssignedVolume[ownerIdx] += calvingThicknessFromThreshold[t,i] * areaCell[i]
+                thresholdBoundarySummedThickness[ownerIdx] += calvingThicknessFromThreshold[t,i]
+                thresholdBoundaryContributors[ownerIdx] += 1
+            #print(thresholdBoundaryAssignedVolume.sum(), (calvingThicknessFromThreshold[t,:]*areaCell[:]).sum())
+            assert np.absolute(thresholdBoundaryAssignedVolume.sum() - (calvingThicknessFromThreshold[t,:]*areaCell[:]).sum()) < 1.0
+            #for i in bdyIndices:
+                #print(f"length={thresholdBoundaryLength[i]}, vol={thresholdBoundaryAssignedVolume[i]}, sumthk={thresholdBoundarySummedThickness[i]}, num={thresholdBoundaryContributors[i]}, meanthk={thresholdBoundarySummedThickness[i]/thresholdBoundaryContributors[i]}")
+            # Finally calculate licalvf for each boundary cell and add to whatever was already there
+            thresholdSpeed = thresholdBoundaryAssignedVolume / \
+                    (thresholdBoundarySummedThickness/thresholdBoundaryContributors * \
+                     thresholdBoundaryLength) / \
+                    deltat[t]    # units of kg / m2 / s
+            # Our estimated threshold speed is really a retreat speed. So to get calving speed, add on the advective speed
+            thresholdFlux[t, bdyIndices] += (thresholdSpeed[bdyIndices] + surfaceSpeed[t,bdyIndices]) * rho_i
+            calvingFluxArray[t,:] += thresholdFlux[t,:]
```

Comment:
```suggestion
            calvingFluxArray[t,bdyIndices] += thresholdFlux[t,bdyIndices]
```
This should amount to the same thing, but I think it's safer than using : indexing.

Reply:
Changed